pkg/graveler/committed/merge_iterator_test.go

```diff
@@ -156,6 +157,359 @@ func TestMerge(t *testing.T) {
+						Count:  int64(len(tst.baseKeys)),
+					}
+				}
+				records := make([]*graveler.ValueRecord, 0)
+				for j, key := range tst.baseKeys[i] {
+					records = append(records, makeV(key, tst.baseIdentities[i][j]))
+				}
+				baseIt.AddRange(rng).
+					AddValueRecords(records...)
+			}
+			// build diff iterator
+			diffIt := testutil.NewFakeDiffIterator()
+			for i, diffRange := range tst.diffRangeIds {
+				var rng *committed.RangeDiff
+				if diffRange != "" {
+					rng = &committed.RangeDiff{
+						Type: tst.diffRangeTypes[i],
+						Range: &committed.Range{
+							ID:     committed.ID(diffRange),
+							MinKey: getMinKey(tst.diffKeys[i]),
+							MaxKey: getMaxKey(tst.diffKeys[i]),
+							Count:  int64(len(tst.diffKeys[i])),
+						},
+					}
+				}
+				records := make([]*graveler.Diff, 0)
+				for j, key := range tst.diffKeys[i] {
+					records = append(records, makeDV(tst.diffRangeTypes[i], key, tst.diffIdentities[i][j], tst.leftIdentities[i][j]))
+				}
+				diffIt.AddRange(rng).
+					AddValueRecords(records...)
+			}
+
+			// test merge iterator
+			ctx := context.Background()
+			it := committed.NewMergeIterator(ctx, diffIt, baseIt)
+			gotKeys := make([]string, 0)
+			gotIDs := make([]string, 0)
+			gotRangesIDs := make([]string, 0)
+			for it.Next() {
+				val, rng := it.Value()
+				gotKeys = append(gotKeys, getRecordKey(val))
+				gotIDs = append(gotIDs, getRecordID(val))
+				gotRangesIDs = append(gotRangesIDs, getRangeID(rng))
+			}
+			if err := it.Err(); err != nil {
+				t.Fatal(err)
+			}
+			if diff := deep.Equal(tst.expectedRanges, gotRangesIDs); diff != nil {
+				t.Errorf("got unexpected ranges from compare iterator. diff=%s", diff)
+			}
+			if diff := deep.Equal(tst.expectedValues, gotKeys); diff != nil {
+				t.Errorf("got unexpected values from merge iterator. diff=%s", diff)
+			}
+			if diff := deep.Equal(tst.expectedIDs, gotIDs); diff != nil {
+				t.Errorf("got unexpected values from merge iterator. diff=%s", diff)
+			}
+		})
+	}
+}
+func TestMergeNextRange(t *testing.T) {
+	baseIt := testutil.NewFakeIterator()
+
+	diffIt := testutil.NewFakeDiffIterator()
+
+	rangeDiff1 := &committed.RangeDiff{
+		Type: added,
+		Range: &committed.Range{
+			ID:     "diff:k1-k3",
+			MinKey: committed.Key("k1"),
+			MaxKey: committed.Key("k3"),
+			Count:  2,
+		},
+	}
+	diffIt.AddRange(rangeDiff1)
+
+	diffIt.AddValueRecords(makeDV(added, "k1", "i1", ""), makeDV(added, "k3", "i3", ""))
+
+	rangeDiff2 := &committed.RangeDiff{
+		Type: added,
+		Range: &committed.Range{
+			ID:     "diff:k4-k5",
+			MinKey: committed.Key("k4"),
+			MaxKey: committed.Key("k5"),
+			Count:  2,
+		},
+	}
+	diffIt.AddRange(rangeDiff2)
+
+	diffIt.AddValueRecords(makeDV(added, "k4", "i4", ""), makeDV(added, "k5", "i5", ""))
+
+	ctx := context.Background()
+	it := committed.NewMergeIterator(ctx, diffIt, baseIt)
+
+	if !it.Next() {
+		t.Fatalf("expected to have next - didn't get get next with error:%v", it.Err())
```

Comment:
2 gets, but also could just be shorter:
```suggestion
		t.Fatalf("expected to have next (err %v)", it.Err())
```